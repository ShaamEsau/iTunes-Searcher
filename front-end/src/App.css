body{
	background-color: powderblue;
}

.App {
  text-align: center;
}

#Logo{
  width: 500px;
  height: 250px;
}

th{
  text-align: center;
}

li{
	list-style: none;
	display: inline;
	padding: 5px;
	border: 3px solid black;

}
