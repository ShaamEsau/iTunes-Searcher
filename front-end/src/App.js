/* Calls the React and Router library to be used */
import React from 'react';
import { HashRouter, Route, NavLink, Switch } from "react-router-dom";
/* Calls stylesheet to edit and style the elements rendered */
import './App.css';
/* Calls image to be used on this page */
import iTunes from "./Images/itunes.jpg";
/* Calls the component to be used on this page */
import Search from "./Components/search.js";
import Favourite from "./Components/favourites.js";

/* Creates a class components */
class App extends React.Component {

    /* Converts data to a readable web format */
    render() {
        /* Data to be called to the web page */
        return (
            /* Creates a div box */
            <div className="App">
            <img id="Logo" src={iTunes}/>
            <br/>
        {/* Placeholder for the component links */}
        <HashRouter>
        <div>
    {/* Creates an unordered list */}
    <ul>
{/* Sets the link and path to the component to be displayed when link is clicked on */}
<li>
<NavLink activeClassName = "active" to= "/Search"> Search </NavLink>
</li>
<li>
<NavLink activeClassName = "active" to = "/Favourite"> Favourites </NavLink>
</li>
</ul>
<Switch>
<Route exact path = {"/Search"} component = {Search} />
<Route path = {"/Favourite"} component = {Favourite} />
</Switch>
</div>
</HashRouter>
</div>
        );
    }
}
/* Allows the component to be imported and used by other file */
export default App;