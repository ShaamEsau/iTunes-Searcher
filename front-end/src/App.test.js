import React from 'react';
import ReactDOM from 'react-dom';
import renderer from "react-test-renderer";
import {shallow, mount} from "enzyme";
import App from './App';
import Search from "./Components/search.js"

/* Initializes a test using the code incased in it  */
it('SnapShot', () => {
	/* Sets a variable of a rendered component */
	const component = renderer.create(<Search/>);
	/* Converts information to JSON */
	let tree = component.toJSON();
	/* Checks is the rendered component matches the snapshot */
	expect(tree).toMatchSnapshot();
});