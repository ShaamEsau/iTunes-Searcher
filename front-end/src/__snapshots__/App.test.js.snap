// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot 1`] = `
<div
  className="App"
>
  <input
    onChange={[Function]}
    type="text"
  />
  <button
    onClick={[Function]}
  >
    Submit
  </button>
  <br />
  <label>
    Song
  </label>
  <input
    name="category"
    onClick={[Function]}
    type="radio"
    value="musicTrack"
  />
  <label>
    Book
  </label>
  <input
    name="category"
    onClick={[Function]}
    type="radio"
    value="ebook"
  />
  <label>
    Audio Book
  </label>
  <input
    name="category"
    onClick={[Function]}
    type="radio"
    value="audiobook"
  />
  <table />
</div>
`;
